contributors/design-proposals/bulk_watch.md

```diff
@@ -0,0 +1,450 @@
+configmaps/... in the system, it is not using watch for this purpose. Instead of
+that, it is retrieving indidual objects, by sending individual GET requests.
+However, to enable automatic updates of mounted secrets/configmaps/..., Kubelet
+is sending those GET requests periodically. In large clusters, this is
+generating huge unnecessary load, as this load in principle should be
+watch-based. We would like to address this together with solving the
+authorization issue.
+
+
+# Proposal
+
+ In this proposal, we're not focusing on how exactly security should be done.
+We're just sketching very high level approach and exact authorization mechanism
+should be discussed separately.
+
+ At the high level, what we would like to achieve is to enable LIST and WATCH
+requests to support more sophisticated filtering (e.g. we would like to be able
+to somehow ask for all Secrets attached to all pods bound to a given node in a
+single LIST or WATCH request). However, the design has to be consistent with
+authorization of other types of requests (in particular GETs).
+
+ To solve this problem, we propose to introduce the idea of `bulk watch ` and
+`bulk list ` (and in the future other bulk operations). This idea was already
+appearing in the community and now we have good usecase to proceed with it.
+
+ Once a bulk watch is set up, we also need to periodically verify its ACLs.
+Whenever a user (in our case Kubelet) loses access to a given resource, the
+watch should be closed within some bounded time. the rationaly behind this
+requirement is that by using (bulk) watch we can't regress the current behavior
+where a polling Kubelet would eventually be unable to access a secret.
+
+ That said, periodic verification of ACLs isn't specific to bulk watch and
+needs to be solved also in `reqular` watch (e.g. user watching just a single
+secret may also lose access to it and such watch should also be closed in this
+case). So this requirement is common for both regular and bulk watch. We
+just need to solve this problem on low enough level that would allow us to
+reuse the same mechanism in both cases - we will solve it by sending an error
+event to the watcher and then just closing this particular watch.
+
+ At the high level, we would like the API to be generic enough so that it will
+be useful for many different usecases, not just this particular one (another
+example may be a controller that needs to deal with just subset of namespaces).
+As a result, below we are describing requirements that the end solution has to
+meet to satisfy our needs
+- a single bulk requests has to support multiple resource types (e.g. get a
+node and all pods associated with it)
+- if the bulk requests filters asks only for object of a single resource type,
+the returned result ideally should be compatible with what is returned from
+the regular API
+- the API has to support aggregation if different resource types are served
```

Comment:
IOW the aggregator needs to implement this, and turn a bulk watch into one or more backing bulk watches. This may be non trivial...